src/coreclr/jit/morph.cpp

```diff
@@ -8755,6 +8755,18 @@ GenTree* Compiler::fgMorphSmpOp(GenTree* tree, MorphAddrContext* mac, bool* optA
             if (fgGlobalMorph && varTypeIsSmall(info.compRetType) && (retVal != nullptr) && !retVal->TypeIs(TYP_VOID) &&
                 fgCastNeeded(retVal, info.compRetType))
             {
+#ifdef SWIFT_SUPPORT
+                // Morph error operand if tree is a GT_SWIFT_ERROR_RET node
+                if (tree->OperIs(GT_SWIFT_ERROR_RET))
+                {
+                    GenTree* const errorVal = fgMorphTree(tree->gtGetOp1());
+                    tree->AsOp()->gtOp1     = errorVal;
+
+                    // Propagate side effect flags
+                    tree->SetAllEffectsFlags(errorVal);
+                }
+#endif // SWIFT_SUPPORT
+
```

Comment:
I think this should be part of the next then branch. Otherwise we'll morph `errorVal` twice when the next branch isn't taken. (It's only needed in case of that early return in the next block; otherwise the fall through path should take care of the morphing)

Reply:
Sorry, I'm not sure what you mean by the next then-block -- this snippet is already on the path of the early return, right?

Reply:
Sorry, I misread the code